llvm/lib/CodeGen/GlobalISel/GISelKnownBits.cpp

```diff
@@ -147,6 +147,15 @@ void GISelKnownBits::computeKnownBitsImpl(Register R, KnownBits &Known,
   unsigned Opcode = MI.getOpcode();
   LLT DstTy = MRI.getType(R);
 
+  // Handle the case where this is called on a register that does not have a
+  // type constraint (i.e. it has a register class constraint instead). This is
```

Comment:
These are not mutually exclusive. A register class constraint does not imply there is no LLT

Reply:
That's true but I'm just moving the existing code in this PR so that we check the LLT is valid before asserting on properties of it. Do you want me to fix that in the same PR?

Reply:
Might as well fix the comment while you're moving it

Reply:
https://github.com/llvm/llvm-project/pull/116083. I can merge that PR into this one if you want to do both at the same time